### PR TITLE
Offers Generation Page & Offers Page

### PR DESCRIPTION
**Pull request by harshitakaushik-dev**

### CODE REVIEW

File: src/components/Card/HomeCard.js
Comment (aindrila-choudhuri) [inline src/components/Card/HomeCard.js:1]: Will you be removing the existing Card component?

File: src/components/Card/HomeCard.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  };
+  return (
+    <TouchableOpacity onPress={handlePress} style={styles.cardContainer}>
+      <Text style={[Styles.mdSemiBold, styles.titleText]}>{data.title}</Text>
+      <View style={styles.rowContainer}>
+        <Text style={styles.infoText}>{data.flats}</Text>
+        <Text style={styles.dot}>•</Text>
+        <Text style={styles.infoText}>{data.distance}</Text>
+      </View>
+      <View style={styles.infoContainer}>
```
Comment (aindrila-choudhuri) [inline src/components/Card/HomeCard.js]: I believe I added a similar comment in the previous PR also, there are lot of duplicate code here, we can use a single component and pass the required parameters

File: src/components/Card/OffersCard.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
+
+import { Styles } from "../../constants/Styles";
+
+const OffersCard = ({ data }) => {
+  let acceptedTextStyle = styles.acceptedText;
+
+  if (data.status !== "Accepted") {
```
Comment (aindrila-choudhuri) [inline src/components/Card/OffersCard.js]: I think we can have two different styles, textStyle and acceptedTextStyle, as per conditions use those styles, rather keeping the styles code in here.

**Review by aindrila-choudhuri — CHANGES_REQUESTED** [review]
Please change the Components name as discussed in the call.

File: src/components/Card/HomeCard.js
Comment (harshitakaushik-dev) [reply src/components/Card/HomeCard.js:1]: Yes

File: src/components/Card/HomeCard.js (hunk 1 above)
Comment (harshitakaushik-dev) [reply src/components/Card/HomeCard.js]: Card will be deleted and HomeCard will remain.

File: src/components/Card/HomeCard.js [hunk 3]
```diff
@@ -3,16 +3,20 @@ import React from "react";
...
     <Pressable
-      onPress={isLoggedIn ? () => navigation.navigate("Offers") : onButtonPress}
+      onPress={
+        isLoggedIn
+          ? () => navigation.navigate("OfferGeneration")
+          : onButtonPress
+      }
       style={styles.cardContainer}
     >
       <Text style={[Styles.mdSemiBold, styles.titleText]}>{data.title}</Text>
```
Comment (aindrila-choudhuri) [inline src/components/Card/HomeCard.js:32]: InfoContainer can be a small component.

File: src/components/Card/OffersCard.js (hunk 2 above)
Comment (aindrila-choudhuri) [reply src/components/Card/OffersCard.js]: Create two styles, and as per condition use them, no need to manipulate styles in code.

File: src/components/Card/OffersRWACard.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import { LocalSvg } from "react-native-svg";
+
+import SelectCampaignCard from "./SelectCampaignCard";
+import SelectDateCard from "./SelectDateCard";
+import { Styles } from "../../constants/Styles";
+
+const OffersRWACard = () => {
+  const [selectedCampaignOption, setSelectedCampaignOption] = useState(null);
+  const [selectedDates, setSelectedDates] = useState([null, null, null]);
+  const [offerButtonColor, setOfferButtonColor] = useState("#557184");
```
Comment (aindrila-choudhuri) [inline src/components/Card/OffersRWACard.js]: Why are we managing styles through component state variables?
Styling should never be component's state's responsibility. 
Use conditional rendering/dynamic class name, etc.

File: src/components/Card/OffersRWACard.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+  useEffect(() => {
```
Comment (aindrila-choudhuri) [inline src/components/Card/OffersRWACard.js]: Remove this logic of maintaining style in useEffect.

File: src/components/Card/OffersRWACard.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  }, [selectedCampaignOption, selectedDates]);
+
+  const navigation = useNavigation();
+
+  const handleOfferClick = () => {
+    if (
+      selectedCampaignOption !== null &&
+      selectedDates.every((date) => date !== null)
+    ) {
+      console.log("Offer clicked!");
```
Comment (aindrila-choudhuri) [inline src/components/Card/OffersRWACard.js]: Remove debug statements.

File: src/components/Card/OffersRWACard.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      navigation.navigate("Offers");
+      console.log(selectedCampaignOption, selectedDates);
```
Comment (aindrila-choudhuri) [inline src/components/Card/OffersRWACard.js]: Remove debug statements.

File: src/screens/SearchPage.js [hunk 8]
```diff
@@ -102,7 +102,7 @@ const styles = StyleSheet.create({
     paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
   },
   searchContainer: {
-    marginTop: 16,
+    // marginTop: 16,
```
Comment (aindrila-choudhuri) [inline src/screens/SearchPage.js]: Remove commented code.

**Review by aindrila-choudhuri — CHANGES_REQUESTED** [review]

File: src/screens/SearchPage.js (hunk 8 above)
Comment (harshitakaushik-dev) [reply src/screens/SearchPage.js]: Removed.

File: src/components/Card/OffersRWACard.js (hunk 7 above)
Comment (harshitakaushik-dev) [reply src/components/Card/OffersRWACard.js]: Removed...

File: src/components/Card/OffersRWACard.js (hunk 6 above)
Comment (harshitakaushik-dev) [reply src/components/Card/OffersRWACard.js]: Removed.

File: src/components/Card/OffersRWACard.js (hunk 4 above)
Comment (harshitakaushik-dev) [reply src/components/Card/OffersRWACard.js]: Updated.

File: src/components/Card/OffersRWACard.js (hunk 5 above)
Comment (harshitakaushik-dev) [reply src/components/Card/OffersRWACard.js]: Updated.

File: src/components/Card/HomeCard.js (hunk 3 above)
Comment (harshitakaushik-dev) [reply src/components/Card/HomeCard.js:32]: HomeCard.js in itself is a small component, and this infoContainer is a non-reusable element. It is different for Offer page as well as Deals Page where different information is displayed in the card. So making a smaller component of it won't help with the reusability or enhancement of the code. If you still think it is a necessity to improve the code, please let me know and I will move ahead with it.

File: src/components/Card/OffersCard.js (hunk 2 above)
Comment (harshitakaushik-dev) [reply src/components/Card/OffersCard.js]: Updated.

File: src/components/Card/HomeCard.js (hunk 3 above)
Comment (aindrila-choudhuri) [reply src/components/Card/HomeCard.js:32]: Here itself it's reusable, in two places we can use it.

**Pull request merged**